com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceOperationsExecution.java

```diff
@@ -43,18 +48,18 @@ public AsyncExecutionState execute(ExecutionWrapper execution) {
 
         Map<CloudServiceExtended, ServiceOperation> servicesWithLastOperation = new HashMap<>();
         for (CloudServiceExtended service : servicesToPoll) {
-            ServiceOperation lastServiceOperation = getLastServiceOperationAndHandleExceptions(execution, client, service);
+            ServiceOperation lastServiceOperation = getLastServiceOperationAndHandleExceptions(execution, service);
             if (lastServiceOperation != null) {
                 servicesWithLastOperation.put(service, lastServiceOperation);
             }
             execution.getStepLogger()
-                .debug(Messages.LAST_OPERATION_FOR_SERVICE, service.getName(), JsonUtil.toJson(lastServiceOperation, true));
+                     .debug(Messages.LAST_OPERATION_FOR_SERVICE, service.getName(), JsonUtil.toJson(lastServiceOperation, true));
         }
-        reportIndividualServiceState(execution, servicesWithLastOperation);
-        reportServiceOperationsState(execution, servicesWithLastOperation, triggeredServiceOperations);
+        reportDetailedServicesStates(execution, servicesWithLastOperation);
+        reportOverallProgress(execution, new ArrayList<>(servicesWithLastOperation.values()), triggeredServiceOperations);
```

Comment:
Arrays.asList?

Reply:
Not possible because Map::values return Collection already